src/cargo/core/shell.rs

```diff
@@ -62,6 +62,18 @@ impl MultiShell {
         MultiShell { out: out, err: err, verbosity: verbosity }
     }
 
+    // Create a quiet, basic shell from supplied writers.
+    pub fn from_write(out: Box<Write + Send>, err: Box<Write + Send>) -> MultiShell {
```

Comment:
We could, it felt like it was exposing a lot of unnecessary details to clients though (need to know about `Shell` and `ShellConfig` and `Verbosity`, where the abstraction here is "send everything to me instead of the console and let me deal with it").